streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -288,7 +288,6 @@ public void run() {
 
             return;
         }
-        setState(RUNNING);
```

Comment:
I could expose the race condition by putting a `sleep(100)` right before this `setState` call, making the tests fail 100%. -- After moving the `setState` (and keeping the sleep) the test passed 100% of the time.